SSLClientESP32 and HTTPUpdate libraries have a conflict

Successfully used SSLClient32 with an experimental  sketch to POST to a Mongodb Database but when merged files with the target application,  discovered some sort of conflict preventing a successful compile.  Reduced everything down to these two libraries. Tried a clean install, same problem.  Used Board definition 'ESP32 Dev Module'
Used the following simple code:

```
#include <SSLClientESP32.h>
#include <HTTPUpdate.h>
void setup() {
  // put your setup code here, to run once:
}
void loop() {
  // put your main code here, to run repeatedly:
}
```

 **Describe the error**
Compile fails at Linking operation.

Could not include compiler output due to Github window 65K size limitations.

**To reproduce it**
Steps to reproduce the behavior:
Compile sketch with ESP32 Dev Module or UM Feathers3 Board definitions.




**Please complete the following information:**
- Hardware: WIndows 10 using Arduinio IDE version 2.1.0  and 2.04
- SSLClientESP32.h version: 2.0.0
- HTTPUpdate.h  version 2.0.0 Arduino included

It appears the email was truncated.    I took the original and cut out everything above the linking. 
Also search for "*********"   I added lines where the compiler output red text   and the line that is the error.Hopefully that helps you see the error lines.

StarDad60


   ----- Forwarded Message ----- From: Kurt Hedlund ***@***.***>To: alkonosst/SSLClientESP32 ***@***.***>Cc: alkonosst/SSLClientESP32 ***@***.***>; Mention ***@***.***>Sent: Wednesday, October 11, 2023 at 07:14:16 PM CDTSubject: Re: [alkonosst/SSLClientESP32] SSLClientESP32 and HTTPUpdate libraries have a conflict (Issue #7)
  Here is my  Compiler output:
StarDad60

#include <SSLClientESP32.h>#include <HTTPUpdate.h>void setup() {  // put your setup code here, to run once:}void loop() {  // put your main code here, to run repeatedly:}



/----cut -----/


/----cut-----/
Linking everything together..."C:\\Users\\kh_admin\\AppData\\Local\\Arduino15\\packages\\esp32\\tools\\xtensa-esp32-elf-gcc\\esp-2021r2-patch5-8.4.0/bin/xtensa-esp32-elf-g++" "-Wl,--Map=C:\\Users\\kh_admin\\AppData\\Local\\Temp\\arduino\\sketches\\BA3B9823BC33BD2D2133733142523E4C/sketch_oct11b.ino.map" "-LC:\\Users\\kh_admin\\AppData\\Local\\Arduino15\\packages\\esp32\\hardware\\esp32\\2.0.9/tools/sdk/esp32/lib" "-LC:\\Users\\kh_admin\\AppData\\Local\\Arduino15\\packages\\esp32\\hardware\\esp32\\2.0.9/tools/sdk/esp32/ld" "-LC:\\Users\\kh_admin\\AppData\\Local\\Arduino15\\packages\\esp32\\hardware\\esp32\\2.0.9/tools/sdk/esp32/qio_qspi" -T esp32.rom.redefined.ld -T memory.ld -T sections.ld -T esp32.rom.ld -T esp32.rom.api.ld -T esp32.rom.libgcc.ld -T esp32.rom.newlib-data.ld -T esp32.rom.syscalls.ld -T esp32.peripherals.ld -mlongcalls -Wno-frame-address -Wl,--cref -Wl,--gc-sections -fno-rtti -fno-lto -Wl,--wrap=esp_log_write -Wl,--wrap=esp_log_writev -Wl,--wrap=log_printf -u ld_include_hli_vectors_bt -u _Z5setupv -u _Z4loopv -u esp_app_desc -u pthread_include_pthread_impl -u pthread_include_pthread_cond_impl -u pthread_include_pthread_local_storage_impl -u pthread_include_pthread_rwlock_impl -u include_esp_phy_override -u ld_include_highint_hdl -u start_app -u start_app_other_cores -u __ubsan_include -Wl,--wrap=longjmp -u __assert_func -u vfs_include_syscalls_impl -Wl,--undefined=uxTopUsedPriority -u app_main -u newlib_include_heap_impl -u newlib_include_syscalls_impl -u newlib_include_pthread_impl -u newlib_include_assert_impl -u __cxa_guard_dummy -DESP32 -DCORE_DEBUG_LEVEL=4 -DARDUINO_RUNNING_CORE=1 -DARDUINO_EVENT_RUNNING_CORE=1 -DARDUINO_USB_CDC_ON_BOOT=0 -Wl,--start-group "C:\\Users\\kh_admin\\AppData\\Local\\Temp\\arduino\\sketches\\BA3B9823BC33BD2D2133733142523E4C\\sketch\\sketch_oct11b.ino.cpp.o" "C:\\Users\\kh_admin\\AppData\\Local\\Temp\\arduino\\sketches\\BA3B9823BC33BD2D2133733142523E4C\\libraries\\SSLClientESP32\\SSLClientESP32.cpp.o" "C:\\Users\\kh_admin\\AppData\\Local\\Temp\\arduino\\sketches\\BA3B9823BC33BD2D2133733142523E4C\\libraries\\SSLClientESP32\\esp_crt_bundle.c.o" "C:\\Users\\kh_admin\\AppData\\Local\\Temp\\arduino\\sketches\\BA3B9823BC33BD2D2133733142523E4C\\libraries\\SSLClientESP32\\ssl_client.cpp.o" "C:\\Users\\kh_admin\\AppData\\Local\\Temp\\arduino\\sketches\\BA3B9823BC33BD2D2133733142523E4C\\libraries\\HTTPUpdate\\HTTPUpdate.cpp.o" "C:\\Users\\kh_admin\\AppData\\Local\\Temp\\arduino\\sketches\\BA3B9823BC33BD2D2133733142523E4C\\libraries\\WiFi\\WiFi.cpp.o" "C:\\Users\\kh_admin\\AppData\\Local\\Temp\\arduino\\sketches\\BA3B9823BC33BD2D2133733142523E4C\\libraries\\WiFi\\WiFiAP.cpp.o" "C:\\Users\\kh_admin\\AppData\\Local\\Temp\\arduino\\sketches\\BA3B9823BC33BD2D2133733142523E4C\\libraries\\WiFi\\WiFiClient.cpp.o" "C:\\Users\\kh_admin\\AppData\\Local\\Temp\\arduino\\sketches\\BA3B9823BC33BD2D2133733142523E4C\\libraries\\WiFi\\WiFiGeneric.cpp.o" "C:\\Users\\kh_admin\\AppData\\Local\\Temp\\arduino\\sketches\\BA3B9823BC33BD2D2133733142523E4C\\libraries\\WiFi\\WiFiMulti.cpp.o" "C:\\Users\\kh_admin\\AppData\\Local\\Temp\\arduino\\sketches\\BA3B9823BC33BD2D2133733142523E4C\\libraries\\WiFi\\WiFiSTA.cpp.o" "C:\\Users\\kh_admin\\AppData\\Local\\Temp\\arduino\\sketches\\BA3B9823BC33BD2D2133733142523E4C\\libraries\\WiFi\\WiFiScan.cpp.o" "C:\\Users\\kh_admin\\AppData\\Local\\Temp\\arduino\\sketches\\BA3B9823BC33BD2D2133733142523E4C\\libraries\\WiFi\\WiFiServer.cpp.o" "C:\\Users\\kh_admin\\AppData\\Local\\Temp\\arduino\\sketches\\BA3B9823BC33BD2D2133733142523E4C\\libraries\\WiFi\\WiFiUdp.cpp.o" "C:\\Users\\kh_admin\\AppData\\Local\\Temp\\arduino\\sketches\\BA3B9823BC33BD2D2133733142523E4C\\libraries\\HTTPClient\\HTTPClient.cpp.o" "C:\\Users\\kh_admin\\AppData\\Local\\Temp\\arduino\\sketches\\BA3B9823BC33BD2D2133733142523E4C\\libraries\\WiFiClientSecure\\WiFiClientSecure.cpp.o" "C:\\Users\\kh_admin\\AppData\\Local\\Temp\\arduino\\sketches\\BA3B9823BC33BD2D2133733142523E4C\\libraries\\WiFiClientSecure\\esp_crt_bundle.c.o" "C:\\Users\\kh_admin\\AppData\\Local\\Temp\\arduino\\sketches\\BA3B9823BC33BD2D2133733142523E4C\\libraries\\WiFiClientSecure\\ssl_client.cpp.o" "C:\\Users\\kh_admin\\AppData\\Local\\Temp\\arduino\\sketches\\BA3B9823BC33BD2D2133733142523E4C\\libraries\\Update\\HttpsOTAUpdate.cpp.o" "C:\\Users\\kh_admin\\AppData\\Local\\Temp\\arduino\\sketches\\BA3B9823BC33BD2D2133733142523E4C\\libraries\\Update\\Updater.cpp.o" "C:\\Users\\kh_admin\\AppData\\Local\\Temp\\arduino\\cores\\e933f5875ffc061bf64497ff7ad85d9d\\core.a" -lesp_ringbuf -lefuse -lesp_ipc -ldriver -lesp_pm -lmbedtls -lapp_update -lbootloader_support -lspi_flash -lnvs_flash -lpthread -lesp_gdbstub -lespcoredump -lesp_phy -lesp_system -lesp_rom -lhal -lvfs -lesp_eth -ltcpip_adapter -lesp_netif -lesp_event -lwpa_supplicant -lesp_wifi -lconsole -llwip -llog -lheap -lsoc -lesp_hw_support -lxtensa -lesp_common -lesp_timer -lfreertos -lnewlib -lcxx -lapp_trace -lasio -lbt -lcbor -lunity -lcmock -lcoap -lnghttp -lesp-tls -lesp_adc_cal -lesp_hid -ltcp_transport -lesp_http_client -lesp_http_server -lesp_https_ota -lesp_https_server -lesp_lcd -lprotobuf-c -lprotocomm -lmdns -lesp_local_ctrl -lsdmmc -lesp_serial_slave_link -lesp_websocket_client -lexpat -lwear_levelling -lfatfs -lfreemodbus -ljsmn -ljson -llibsodium -lmqtt -lopenssl -lperfmon -lspiffs -lulp -lwifi_provisioning -lrmaker_common -lesp_diagnostics -lrtc_store -lesp_insights -ljson_parser -ljson_generator -lesp_schedule -lespressif__esp_secure_cert_mgr -lesp_rainmaker -lgpio_button -lqrcode -lws2812_led -lesp-sr -lesp32-camera -lesp_littlefs -lespressif__esp-dsp -lfb_gfx -lasio -lcmock -lunity -lcoap -lesp_lcd -lesp_websocket_client -lexpat -lfreemodbus -ljsmn -llibsodium -lperfmon -lesp_adc_cal -lesp_hid -lfatfs -lwear_levelling -lopenssl -lesp_insights -lcbor -lesp_diagnostics -lrtc_store -lesp_rainmaker -lesp_local_ctrl -lesp_https_server -lwifi_provisioning -lprotocomm -lbt -lbtdm_app -lprotobuf-c -lmdns -ljson_parser -ljson_generator -lesp_schedule -lespressif__esp_secure_cert_mgr -lqrcode -lrmaker_common -lmqtt -lcat_face_detect -lhuman_face_detect -lcolor_detect -lmfn -ldl -lmultinet -lesp_audio_processor -lesp_audio_front_end -lwakenet -lesp-sr -lmultinet -lesp_audio_processor -lesp_audio_front_end -lwakenet -ljson -lspiffs -ldl_lib -lc_speech_features -lwakeword_model -lmultinet2_ch -lesp_tts_chinese -lvoice_set_xiaole -lesp_ringbuf -lefuse -lesp_ipc -ldriver -lesp_pm -lmbedtls -lapp_update -lbootloader_support -lspi_flash -lnvs_flash -lpthread -lesp_gdbstub -lespcoredump -lesp_phy -lesp_system -lesp_rom -lhal -lvfs -lesp_eth -ltcpip_adapter -lesp_netif -lesp_event -lwpa_supplicant -lesp_wifi -lconsole -llwip -llog -lheap -lsoc -lesp_hw_support -lxtensa -lesp_common -lesp_timer -lfreertos -lnewlib -lcxx -lapp_trace -lnghttp -lesp-tls -ltcp_transport -lesp_http_client -lesp_http_server -lesp_https_ota -lsdmmc -lesp_serial_slave_link -lulp -lmbedtls_2 -lmbedcrypto -lmbedx509 -lcoexist -lcore -lespnow -lmesh -lnet80211 -lpp -lsmartconfig -lwapi -lesp_ringbuf -lefuse -lesp_ipc -ldriver -lesp_pm -lmbedtls -lapp_update -lbootloader_support -lspi_flash -lnvs_flash -lpthread -lesp_gdbstub -lespcoredump -lesp_phy -lesp_system -lesp_rom -lhal -lvfs -lesp_eth -ltcpip_adapter -lesp_netif -lesp_event -lwpa_supplicant -lesp_wifi -lconsole -llwip -llog -lheap -lsoc -lesp_hw_support -lxtensa -lesp_common -lesp_timer -lfreertos -lnewlib -lcxx -lapp_trace -lnghttp -lesp-tls -ltcp_transport -lesp_http_client -lesp_http_server -lesp_https_ota -lsdmmc -lesp_serial_slave_link -lulp -lmbedtls_2 -lmbedcrypto -lmbedx509 -lcoexist -lcore -lespnow -lmesh -lnet80211 -lpp -lsmartconfig -lwapi -lesp_ringbuf -lefuse -lesp_ipc -ldriver -lesp_pm -lmbedtls -lapp_update -lbootloader_support -lspi_flash -lnvs_flash -lpthread -lesp_gdbstub -lespcoredump -lesp_phy -lesp_system -lesp_rom -lhal -lvfs -lesp_eth -ltcpip_adapter -lesp_netif -lesp_event -lwpa_supplicant -lesp_wifi -lconsole -llwip -llog -lheap -lsoc -lesp_hw_support -lxtensa -lesp_common -lesp_timer -lfreertos -lnewlib -lcxx -lapp_trace -lnghttp -lesp-tls -ltcp_transport -lesp_http_client -lesp_http_server -lesp_https_ota -lsdmmc -lesp_serial_slave_link -lulp -lmbedtls_2 -lmbedcrypto -lmbedx509 -lcoexist -lcore -lespnow -lmesh -lnet80211 -lpp -lsmartconfig -lwapi -lesp_ringbuf -lefuse -lesp_ipc -ldriver -lesp_pm -lmbedtls -lapp_update -lbootloader_support -lspi_flash -lnvs_flash -lpthread -lesp_gdbstub -lespcoredump -lesp_phy -lesp_system -lesp_rom -lhal -lvfs -lesp_eth -ltcpip_adapter -lesp_netif -lesp_event -lwpa_supplicant -lesp_wifi -lconsole -llwip -llog -lheap -lsoc -lesp_hw_support -lxtensa -lesp_common -lesp_timer -lfreertos -lnewlib -lcxx -lapp_trace -lnghttp -lesp-tls -ltcp_transport -lesp_http_client -lesp_http_server -lesp_https_ota -lsdmmc -lesp_serial_slave_link -lulp -lmbedtls_2 -lmbedcrypto -lmbedx509 -lcoexist -lcore -lespnow -lmesh -lnet80211 -lpp -lsmartconfig -lwapi -lesp_ringbuf -lefuse -lesp_ipc -ldriver -lesp_pm -lmbedtls -lapp_update -lbootloader_support -lspi_flash -lnvs_flash -lpthread -lesp_gdbstub -lespcoredump -lesp_phy -lesp_system -lesp_rom -lhal -lvfs -lesp_eth -ltcpip_adapter -lesp_netif -lesp_event -lwpa_supplicant -lesp_wifi -lconsole -llwip -llog -lheap -lsoc -lesp_hw_support -lxtensa -lesp_common -lesp_timer -lfreertos -lnewlib -lcxx -lapp_trace -lnghttp -lesp-tls -ltcp_transport -lesp_http_client -lesp_http_server -lesp_https_ota -lsdmmc -lesp_serial_slave_link -lulp -lmbedtls_2 -lmbedcrypto -lmbedx509 -lcoexist -lcore -lespnow -lmesh -lnet80211 -lpp -lsmartconfig -lwapi -lphy -lrtc -lesp_phy -lphy -lrtc -lesp_phy -lphy -lrtc -lxt_hal -lm -lnewlib -lstdc++ -lpthread -lgcc -lcxx -lapp_trace -lgcov -lapp_trace -lgcov -lc -Wl,--end-group -Wl,-EL -o "C:\\Users\\kh_admin\\AppData\\Local\\Temp\\arduino\\sketches\\BA3B9823BC33BD2D2133733142523E4C/sketch_oct11b.ino.elf"



********************FROM THIS POINT DOWN the COMPILER colors the text RED********************************************************************c:/users/kh_admin/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/esp-2021r2-patch5-8.4.0/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld.exe: C:\Users\kh_admin\AppData\Local\Temp\arduino\sketches\BA3B9823BC33BD2D2133733142523E4C\libraries\WiFiClientSecure\esp_crt_bundle.c.o: in function `arduino_esp_crt_bundle_attach':C:\Users\kh_admin\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.9\libraries\WiFiClientSecure\src/esp_crt_bundle.c:180: multiple definition of `arduino_esp_crt_bundle_attach'; C:\Users\kh_admin\AppData\Local\Temp\arduino\sketches\BA3B9823BC33BD2D2133733142523E4C\libraries\SSLClientESP32\esp_crt_bundle.c.o:c:\Users\kh_admin\Documents\Arduino\libraries\SSLClientESP32\src/esp_crt_bundle.c:180: first defined herec:/users/kh_admin/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/esp-2021r2-patch5-8.4.0/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld.exe: C:\Users\kh_admin\AppData\Local\Temp\arduino\sketches\BA3B9823BC33BD2D2133733142523E4C\libraries\WiFiClientSecure\esp_crt_bundle.c.o: in function `arduino_esp_crt_bundle_detach':C:\Users\kh_admin\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.9\libraries\WiFiClientSecure\src/esp_crt_bundle.c:203: multiple definition of `arduino_esp_crt_bundle_detach'; C:\Users\kh_admin\AppData\Local\Temp\arduino\sketches\BA3B9823BC33BD2D2133733142523E4C\libraries\SSLClientESP32\esp_crt_bundle.c.o:c:\Users\kh_admin\Documents\Arduino\libraries\SSLClientESP32\src/esp_crt_bundle.c:203: first defined herec:/users/kh_admin/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/esp-2021r2-patch5-8.4.0/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld.exe: C:\Users\kh_admin\AppData\Local\Temp\arduino\sketches\BA3B9823BC33BD2D2133733142523E4C\libraries\WiFiClientSecure\esp_crt_bundle.c.o: in function `arduino_esp_crt_bundle_set':C:\Users\kh_admin\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.9\libraries\WiFiClientSecure\src/esp_crt_bundle.c:212: multiple definition of `arduino_esp_crt_bundle_set';
**************************THE FOLLOWING LINE SEEMS TO BE THE ERROR LINE ************************************************************************ C:\Users\kh_admin\AppData\Local\Temp\arduino\sketches\BA3B9823BC33BD2D2133733142523E4C\libraries\SSLClientESP32\esp_crt_bundle.c.o:c:\Users\kh_admin\Documents\Arduino\libraries\SSLClientESP32\src/esp_crt_bundle.c:212: first defined herecollect2.exe: error: ld returned 1 exit status
Using library SSLClientESP32 at version 2.0.0 in folder: C:\Users\kh_admin\Documents\Arduino\libraries\SSLClientESP32 Using library HTTPUpdate at version 2.0.0 in folder: C:\Users\kh_admin\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.9\libraries\HTTPUpdate Using library WiFi at version 2.0.0 in folder: C:\Users\kh_admin\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.9\libraries\WiFi Using library HTTPClient at version 2.0.0 in folder: C:\Users\kh_admin\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.9\libraries\HTTPClient Using library WiFiClientSecure at version 2.0.0 in folder: C:\Users\kh_admin\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.9\libraries\WiFiClientSecure Using library Update at version 2.0.0 in folder: C:\Users\kh_admin\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.9\libraries\Update exit status 1
Compilation error: exit status 1














Message ID: ***@***.***>

I think I found the errors. There are some definition conflicts with `WiFiClientSecure` library of ESP32 core:
> multiple definition of `arduino_esp_crt_bundle_attach'

> multiple definition of `arduino_esp_crt_bundle_detach'

> multiple definition of `arduino_esp_crt_bundle_set'

I fixed these errors in a new branch (https://github.com/alkonosst/SSLClientESP32/tree/fix-definition-conflict-with-wificlientsecure). Can you please test this?

Excellent!   Looks good.
Compiled without failure.   I don't have active programs that use both because of the conflict, I found an alternative. I do have active programs  that use the SSLClientESP32 library.Re-compiled a couple of  those and checked functionality, that looks good as well.
Thanks,
 Stadad60
  

    On Friday, October 13, 2023 at 09:55:13 AM CDT, Maximiliano Ramirez ***@***.***> wrote:  
 
 


I think I found the errors. There are some definition conflicts with WiFiClientSecure library of ESP32 core:


multiple definition of `arduino_esp_crt_bundle_attach'



multiple definition of `arduino_esp_crt_bundle_detach'



multiple definition of `arduino_esp_crt_bundle_set'


I fixed these errors in a new branch (https://github.com/alkonosst/SSLClientESP32/tree/fix-definition-conflict-with-wificlientsecure). Can you please test this?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>

--- a/examples/https_wifi_and_sim7600_certbundle/src/main.cpp
+++ b/examples/https_wifi_and_sim7600_certbundle/src/main.cpp
@@ -8,7 +8,7 @@
 #define TINY_GSM_RX_BUFFER 1024
 #define TINY_GSM_DEBUG     Serial
 
-#include "SSLClient.h"
+#include "SSLClientESP32.h"
 #include "TinyGSM.h"
 #include <Arduino.h>
 #include <WiFi.h>
@@ -36,7 +36,7 @@ WiFiClient client_wifi;
 TinyGsmClient client_modem(modem);
 
 // * Secure client object, initialized with Wifi client
-SSLClient ssl(&client_wifi);
+SSLClientESP32 ssl(&client_wifi);
 
 // * Certificate bundle with 41 most used root certificates
 extern const uint8_t ca_cert_bundle_start[] asm("_binary_data_crt_x509_crt_bundle_bin_start");

--- a/src/SSLClientESP32.cpp
+++ b/src/SSLClientESP32.cpp
@@ -18,7 +18,7 @@
 */
 
 #include "SSLClientESP32.h"
-#include "esp_crt_bundle.h"
+#include "ssl_lib_crt_bundle.h"
 #include <errno.h>
 
 #undef connect
@@ -257,10 +257,10 @@ void SSLClientESP32::setCACertBundle(const uint8_t * bundle)
 {
     if (bundle != NULL)
     {
-        arduino_esp_crt_bundle_set(bundle);
+        ssl_lib_crt_bundle_set(bundle);
         _use_ca_bundle = true;
     } else {
-        arduino_esp_crt_bundle_detach(NULL);
+        ssl_lib_crt_bundle_detach(NULL);
         _use_ca_bundle = false;
     }
 }

--- a/src/SSLClientESP32.h
+++ b/src/SSLClientESP32.h
@@ -17,11 +17,12 @@
   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 */
 
-#ifndef SSLClientESP32_H
-#define SSLClientESP32_H
+#ifndef SSLCLIENTESP32_H
+#define SSLCLIENTESP32_H
+
 #include "Arduino.h"
 #include "IPAddress.h"
-#include "ssl_client.h"
+#include "ssl_lib_client.h"
 
 class SSLClientESP32 : public Client
 {
@@ -101,4 +102,4 @@ private:
     using Print::write;
 };
 
-#endif /* SSLClientESP32_H */
+#endif /* SSLCLIENTESP32_H */

--- a/src/ssl_lib_client.cpp
+++ b/src/ssl_lib_client.cpp
@@ -13,8 +13,8 @@
 #include <mbedtls/oid.h>
 #include <algorithm>
 #include <string>
-#include "ssl_client.h"
-#include "esp_crt_bundle.h"
+#include "ssl_lib_client.h"
+#include "ssl_lib_crt_bundle.h"
 
 namespace SSLClientLib {
 
@@ -214,7 +214,7 @@ int start_ssl_client(sslclient_context *ssl_client, const char *host, uint32_t p
         }
     } else if (useRootCABundle) {
         log_v("Attaching root CA cert bundle");
-        ret = arduino_esp_crt_bundle_attach(&ssl_client->ssl_conf);
+        ret = ssl_lib_crt_bundle_attach(&ssl_client->ssl_conf);
 
         if (ret < 0) {
             return handle_error(ret);

--- a/src/ssl_lib_client.h
+++ b/src/ssl_lib_client.h
@@ -4,8 +4,9 @@
  * Additions Copyright (C) 2023 Maximiliano Ramirez.
  */
 
-#ifndef SSLCLIENT_LIB_ARD_SSL_H
-#define SSLCLIENT_LIB_ARD_SSL_H
+#ifndef SSL_LIB_ARD_SSL_H
+#define SSL_LIB_ARD_SSL_H
+
 #include "mbedtls/platform.h"
 #include "mbedtls/net.h"
 #include "mbedtls/debug.h"
@@ -47,4 +48,4 @@ bool get_peer_fingerprint(sslclient_context *ssl_client, uint8_t sha256[32]);
 
 }
 
-#endif
+#endif // SSL_LIB_ARD_SSL_H

--- a/src/ssl_lib_crt_bundle.c
+++ b/src/ssl_lib_crt_bundle.c
@@ -16,8 +16,7 @@
 #include <string.h>
 #include <esp_system.h>
 #include <esp32-hal-log.h>
-#include "esp_crt_bundle.h"
-#include "esp_err.h"
+#include "ssl_lib_crt_bundle.h"
 
 #define BUNDLE_HEADER_OFFSET 2
 #define CRT_HEADER_OFFSET 4
@@ -176,7 +175,7 @@ static esp_err_t esp_crt_bundle_init(const uint8_t *x509_bundle)
     return ESP_OK;
 }
 
-esp_err_t arduino_esp_crt_bundle_attach(void *conf)
+esp_err_t ssl_lib_crt_bundle_attach(void *conf)
 {
     esp_err_t ret = ESP_OK;
     // If no bundle has been set by the user then use the bundle embedded in the binary
@@ -199,7 +198,7 @@ esp_err_t arduino_esp_crt_bundle_attach(void *conf)
     return ret;
 }
 
-void arduino_esp_crt_bundle_detach(mbedtls_ssl_config *conf)
+void ssl_lib_crt_bundle_detach(mbedtls_ssl_config *conf)
 {
     free(s_crt_bundle.crts);
     s_crt_bundle.crts = NULL;
@@ -208,7 +207,7 @@ void arduino_esp_crt_bundle_detach(mbedtls_ssl_config *conf)
     }
 }
 
-void arduino_esp_crt_bundle_set(const uint8_t *x509_bundle)
+void ssl_lib_crt_bundle_set(const uint8_t *x509_bundle)
 {
     // Free any previously used bundle
     free(s_crt_bundle.crts);

--- a/src/ssl_lib_crt_bundle.h
+++ b/src/ssl_lib_crt_bundle.h
@@ -13,10 +13,11 @@
 // limitations under the License.
 
 
-#ifndef _ESP_CRT_BUNDLE_H_
-#define _ESP_CRT_BUNDLE_H_
+#ifndef SSL_LIB_CRT_BUNDLE_H
+#define SSL_LIB_CRT_BUNDLE_H
 
 #include "mbedtls/ssl.h"
+#include "esp_err.h"
 
 #ifdef __cplusplus
 extern "C" {
@@ -36,7 +37,7 @@ extern "C" {
  *             - ESP_OK  if adding certificates was successful.
  *             - Other   if an error occured or an action must be taken by the calling process.
  */
-esp_err_t arduino_esp_crt_bundle_attach(void *conf);
+esp_err_t ssl_lib_crt_bundle_attach(void *conf);
 
 
 /**
@@ -46,7 +47,7 @@ esp_err_t arduino_esp_crt_bundle_attach(void *conf);
  *
  * @param[in]  conf      The config struct for the SSL connection.
  */
-void arduino_esp_crt_bundle_detach(mbedtls_ssl_config *conf);
+void ssl_lib_crt_bundle_detach(mbedtls_ssl_config *conf);
 
 
 /**
@@ -58,11 +59,11 @@ void arduino_esp_crt_bundle_detach(mbedtls_ssl_config *conf);
  *
  * @param[in]  x509_bundle     A pointer to the certificate bundle.
  */
-void arduino_esp_crt_bundle_set(const uint8_t *x509_bundle);
+void ssl_lib_crt_bundle_set(const uint8_t *x509_bundle);
 
 
 #ifdef __cplusplus
 }
 #endif
 
-#endif //_ESP_CRT_BUNDLE_H_
+#endif //SSL_LIB_CRT_BUNDLE_H

